File: reader/server/core/backend/src/main/java/com/richodemus/reader/backend/feed/FeedRepository.java
package com.richodemus.reader.backend.feed;

import com.richodemus.reader.backend.model.Feed;
import com.richodemus.reader.dto.FeedId;
import com.richodemus.reader.dto.FeedUrl;

import java.util.Optional;

public interface FeedRepository {
    Optional<Feed> getFeed(FeedId feedId);

    FeedId getFeedId(FeedUrl feedUrl);
}

File: reader/server/core/backend/src/main/kotlin/com/richodemus/reader/backend/feed/FeedRepository.kt
package com.richodemus.reader.backend.feed

import com.richodemus.reader.backend.model.Feed
import com.richodemus.reader.dto.FeedId
import com.richodemus.reader.dto.FeedUrl

interface FeedRepository {
    fun getFeed(feedId: FeedId): Feed?
    fun getFeedId(feedUrl: FeedUrl): FeedId?
}
